metaflow/plugins/cards/card_modules/__init__.py

```diff
@@ -14,53 +16,39 @@ def iter_namespace(ns_pkg):
     return pkgutil.iter_modules(ns_pkg.__path__, ns_pkg.__name__ + ".")
 
 
-def _get_external_card_packages(with_paths=False):
+def _get_external_card_packages():
     """
-    Safely extract all exteranl card modules
-    Args:
-        with_paths (bool, optional): setting `with_paths=True` will result in a list of tuples: `[( mf_extensions_parent_path , relative_path_to_module, module)]`. setting false will return a list of modules Defaults to False.
-
+    Safely extract all external card modules
     Returns:
-        `list` of `ModuleType` or `list` of `tuples` where each tuple if of the form (mf_extensions_parent_path , relative_path_to_module, ModuleType)
+        `list` of `ModuleType`
     """
     import importlib
 
-    available_card_modules = []
+    # Caching card related modules.
+    global _CARD_MODULES
+    if len(_CARD_MODULES) > 0:
+        return _CARD_MODULES
     for m in get_modules("plugins.cards"):
         # Iterate submodules of metaflow_extensions.X.plugins.cards
         # For example metaflow_extensions.X.plugins.cards.monitoring
         card_packages = []
-        for fndx, card_mod, ispkg_c in iter_namespace(m.module):
+        if getattr(m, "__path__", None):
```

Comment:
There are three cases we want to support (I think):
- your current system where you have a namespace package and then sub packages that have `__init__.py`
- a namespace package support with `mfextinit_X.py`
- a regular package support

In the last two cases, you can just do the `append(m)` with no issues. You can check to see if you have `__file__` as an attribute of `m.module` (not `m`). I would then put an `else` and do the `for` loop.